prow/pod-utils/gcs/upload.go

```diff
@@ -150,47 +152,73 @@ func FileUploadWithOptions(file string, opts pkgio.WriterOptions) UploadFunc {
 			}
 		}
 
-		uploadErr := DataUploadWithOptions(reader, opts)(writer)
+		firstTimeNewReader := true
```

Comment:
should this be above the `return` line? My understanding is that when this function is retried, this line `firstTimeNewReader := true` is executed every time, so it'll never be false.